super-csv/src/test/java/org/supercsv/io/TokenizerTest.java

```diff
@@ -586,6 +587,178 @@ public void testQuotedSpaces() throws Exception {
 	}
+
+	/**
+	 * Tests the readColumns() method with a quote character escaped with another quote (RFC style)
+	 */
+	@Test
+	public void testQuotedQuoteCharWithDoubleQuoteEscape() throws Exception {
+		final String input = "one,\"two\",\"field with \"\" quote char\",four";
+		tokenizer = createTokenizer(input, NORMAL_PREFERENCE);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 4);
+		assertEquals("one", columns.get(0));
+		assertEquals("two", columns.get(1));
+		assertEquals("field with \" quote char", columns.get(2));
+		assertEquals("four", columns.get(3));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with a quote character escaped with another quote (RFC style)
+	 */
+	@Test
+	public void testQuotedQuoteCharWithBackslashEscape() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "one,two,\"field with \\\" quote char\",four";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 4);
+		assertEquals("one", columns.get(0));
+		assertEquals("two", columns.get(1));
+		assertEquals("field with \" quote char", columns.get(2));
+		assertEquals("four", columns.get(3));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an escape character following another escape char.
+	 */
+	@Test
+	public void testEscapedEscapeChar() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"field with \\\\ escape char\"";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 1);
+		assertEquals("field with \\ escape char", columns.get(0));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an even number of escape characters
+	 * char.
+	 */
+	@Test
+	public void testEvenSeriesOfEscapeChars() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"\\\\\\\\\\\\\"";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 1);
+		assertEquals("\\\\\\", columns.get(0));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an odd number of escape characters
+	 * char.
+	 */
+	@Test(expected = SuperCsvException.class)
+	public void testOddSeriesOfEscapeChars() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"\\\\\\\\\\\"";
```

Comment:
The `"` there are required to make this a quoted field.  Trickiness with Java literals..  what that evaluates to is a literal String: `"\\\\\"` which has an odd number of quoted fields.

I've seen problems in the past with poorly-written lookahead parsers where a long string series of "special" characters caused problems so wanted to ensure that didn't affect this parser implementation.  I can drop to 3 if you'd like.